soc/arm/ti_k3/j721e/soc.c

```diff
@@ -0,0 +1,60 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/fatal.h>
+
+#include <zephyr/drivers/interrupt_controller/intc_vim.h>
+
+static int j721e_init(void)
+{
+	return 0;
+}
```

Comment:
remove if not needed

Reply:
+1, drop if empty

Reply:
Can we pass NULL as a first argument to SYS_INIT()?

Reply:
what's the point of a no-op SYS_INIT...?

Reply:
Removing SYS_INIT() and j721e_init().

Reply:
@slpp95prashanth , sorry for the late response, but you will need to atleast unlock the CTRL MMR regions during, all the necessary regions might not always be unlocked by bootloader for you.
See https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/ti_k3/am62x_m4/soc.c#L47